src/databricks/labs/remorph/reconcile/query_builder/threshold_query.py

```diff
@@ -29,3 +50,69 @@ def build_query(self) -> str:
+        from_clause, join_clause = self._generate_from_and_join_clause()
+        return select(*select_clause).from_(from_clause).join(join_clause).where(where).sql(dialect=Databricks)
+
+    def _generate_select_where_clause(self) -> tuple[list[exp.Expression], exp.Expression]:
+        thresholds = self.table_conf.thresholds
+
+        def _build_alias(this, table_name):
+            return build_alias(this=this, alias=f"{this}_{table_name}", table_name=table_name)
+
+        def _build_absolute_case(base, threshold, column):
+            return build_alias(build_threshold_absolute_case(base=base, threshold=threshold), alias=f"{column}_match")
+
+        select_clause = []
+        where_clause = []
+
+        for threshold in thresholds:
+            column = threshold.column_name
+            source_col = _build_alias(column, "source")
+            databricks_col = _build_alias(column, "databricks")
+
+            base = exp.Paren(
+                this=build_sub(
+                    left_column_name=column,
+                    left_table_name="source",
+                    right_column_name=column,
+                    right_table_name="databricks",
+                )
+            ).transform(coalesce)
+
+            if threshold.get_type() == ThresholdMode.NUMBER_ABSOLUTE.value:
+                select_clause.append(source_col.transform(coalesce))
+                select_clause.append(databricks_col.transform(coalesce))
+                select_clause.append(_build_absolute_case(base, threshold, column))
+                where_clause.append(exp.NEQ(this=base, expression=exp.Literal(this="0", is_string=False)))
+            elif threshold.get_type() == ThresholdMode.NUMBER_PERCENTILE.value:
+                select_clause.append(source_col.transform(coalesce))
+                select_clause.append(databricks_col.transform(coalesce))
+                this = build_threshold_percentile_case(base=base, threshold=threshold)
+                select_clause.append(build_alias(this=this, alias=f"{column}_match"))
+                where_clause.append(exp.NEQ(this=base, expression=exp.Literal(this="0", is_string=False)))
+            else:
+                select_clause.append(source_col.transform(anonymous, "unix_timestamp").transform(coalesce))
+                select_clause.append(databricks_col.transform(anonymous, "unix_timestamp").transform(coalesce))
+                exp_anonymous = base.transform(anonymous, "unix_timestamp")
+                select_clause.append(_build_absolute_case(exp_anonymous, threshold, column))
+                where_clause.append(exp.NEQ(this=exp_anonymous, expression=exp.Literal(this="0", is_string=False)))
+
+        for column in sorted(self.table_conf.get_join_columns):
+            select_clause.append(_build_alias(column, "source"))
+        where = build_where_clause(where_clause)
+
+        return select_clause, where
+
+    def _generate_from_and_join_clause(self) -> tuple[From, Join]:
```

Comment:
let us be consistent in representing expressions, I see ```exp.Column``` as well as ```From``` I would prefer former

Reply:
Done. referring all the expression as exp.